[BUG] Hooks plugin with `on_env` & jinja filter : TemplateAssertionError

- Python version : 3.8
- Plugin version : 0.53

Configuration : 
```
plugins:
  - search
  - ezlinks:
        wikilinks: true
  - mermaid2
  - awesome-pages:
      collapse_single_pages: true
  - embed_file:
      callouts: true
      custom-attributes: 'assets/css/custom_attributes.css'
      language_message: 'ERROR 404 : NOT FOUND'
  - callouts
  - custom-attributes:
      file: 'assets/css/custom_attributes.css'
  - meta-descriptions
  - i18n:
      default_language: en
      docs_structure: folder
      languages:
        en:
          name: English
        fr:
          name: Français
extra_javascript:
  - assets/js/mathjax.js
  - assets/js/utils.js
  - assets/js/wiki_hover.js
  - https://polyfill.io/v3/polyfill.min.js?features=es6
  - https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js
hooks:
  - overrides/hooks/on_page_markdown.py
  - overrides/hooks/on_env.py
  ```

Removing i18n make the build successfull.

Log : 
```
ERROR    -  Error building page 'en/index.md': No filter named 'value_in_frontmatter'.
Traceback (most recent call last):
  File "C:\Users\lilis\anaconda3\envs\Publisher\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\lilis\anaconda3\envs\Publisher\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\lilis\anaconda3\envs\Publisher\Scripts\mkdocs.exe\__main__.py", line 7, in <module>
  File "C:\Users\lilis\anaconda3\envs\Publisher\lib\site-packages\click\core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "C:\Users\lilis\anaconda3\envs\Publisher\lib\site-packages\click\core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "C:\Users\lilis\anaconda3\envs\Publisher\lib\site-packages\click\core.py", line 1657, in invoke
    {{ super() }}
  File "C:\Users\lilis\anaconda3\envs\Publisher\lib\site-packages\material\base.html", line 178, in block 'content'
    {% include "partials/content.html" %}
  File "C:\Users\lilis\anaconda3\envs\Publisher\lib\site-packages\material\partials\content.html", line 20, in top-level template code
    {% include "partials/comments.html" %}
  File "C:\Users\lilis\anaconda3\envs\Publisher\lib\site-packages\jinja2\environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "D:\Documents\Github\obsidian_mkdocs_publisher_docs\overrides\partials\comments.html", line 1, in template
    {% set local_comments = "comments" | value_in_frontmatter(page.meta) %}
jinja2.exceptions.TemplateAssertionError: No filter named 'value_in_frontmatter'.
```

Steps : 
1. Write a hooks (here it's in `overrides/hooks/on_env.py`) modifying template filter ([here is my file](https://github.com/ObsidianPublisher/publisher-template-gh-pages/blob/main/overrides/hooks/on_env.py))
2. Add the hooks in the `hooks` part of the mkdocs.yml
3. Run `mkdocs serve`

Hi @Lisandra-dev 

I'm trying to understand what happens. I'm puzzled because if I pass the wrong arguments to the filter I get a contextual error 

```
    {% set test = "comments" | value_in_frontmatter() %}
    E   TypeError: value_in_frontmatter() missing 1 required positional argument: 'metadata'
```

So it looks like it knows the function filter... only when we pass the correct arguments we get the unknown filter error...

--- a/mkdocs_static_i18n/plugin.py
+++ b/mkdocs_static_i18n/plugin.py
@@ -4,6 +4,7 @@ from copy import deepcopy
 from pathlib import Path
 
 from mkdocs import __version__ as mkdocs_version
+from mkdocs import plugins
 from mkdocs.commands.build import _build_page, _populate_page
 from mkdocs.config.config_options import Choice, Type
 from mkdocs.plugins import BasePlugin
@@ -460,6 +461,20 @@ class I18n(BasePlugin):
                     )
                     search_plugin.search_index._entries.remove(duplicated_entry)
 
+    @plugins.event_priority(-100)
+    def on_env(self, env, config, files, **kwargs):
+        """
+        Copy the main env because it can have been altered by hooks.
+
+        Since hooks are injected as plugins, we need to make sure to
+        run last using event_priority.
+
+        See #178.
+        """
+        for language in self.config["languages"].keys():
+            self.i18n_configs[language]["env"] = env
+        return env
+
     def on_page_markdown(self, markdown, page, config, files):
         """
         Use the 'page_markdown' event to translate page titles as well
@@ -557,7 +572,7 @@ class I18n(BasePlugin):
             log.info(f"Building {language} documentation")
 
             config = self.i18n_configs[language]
-            env = self.i18n_configs[language]["theme"].get_env()
+            env = self.i18n_configs[language]["env"]
             files = self.i18n_files[language]
             nav = self.i18n_navs[language]
 

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,2 +1,2 @@
-mkdocs>=1.2.1
+mkdocs>=1.4.2
 mkdocs-material==8.5.10

--- a/setup.py
+++ b/setup.py
@@ -25,7 +25,7 @@ setup(
     platforms="any",
     packages=find_packages(),
     include_package_data=True,
-    install_requires=["mkdocs>=1.2.3"],
+    install_requires=["mkdocs>=1.4.2"],
     entry_points={"mkdocs.plugins": ["i18n = mkdocs_static_i18n.plugin:I18n"]},
     python_requires=">=3.7",
     classifiers=[

--- a/tests/hooks_jinja_on_env.py
+++ b/tests/hooks_jinja_on_env.py
@@ -1,0 +1,19 @@
+def value_in_frontmatter(key, metadata):
+    """Check if a key exists in the frontmatter.
+
+    Args:
+        key (any): the key to check
+        metadata (any): the frontmatter
+
+    Returns:
+        bool: true if exists
+    """
+    if key in metadata:
+        return metadata[key]
+    else:
+        return None
+
+
+def on_env(env, config, files, **kwargs):
+    env.filters["value_in_frontmatter"] = value_in_frontmatter
+    return env

--- a/tests/test_hooks.py
+++ b/tests/test_hooks.py
@@ -19,3 +19,23 @@ def test_hooks_working():
         hooks=["hooks.py"],
     )
     build(mkdocs_config)
+
+
+def test_hooks_env_modified():
+    mkdocs_config = load_config(
+        "tests/mkdocs_base.yml",
+        theme={"name": "mkdocs", "custom_dir": "theme_overrides"},
+        use_directory_urls=True,
+        docs_dir="docs_suffix_structure/",
+        site_url="http://localhost",
+        extra_javascript=[],
+        plugins={
+            "i18n": {
+                "default_language": "en",
+                "languages": {"fr": "français", "en": "english"},
+            },
+        },
+        hooks=["hooks_jinja_on_env.py"],
+        site_dir="/tmp/xoxo/",
+    )
+    build(mkdocs_config)

--- a/tests/test_navigation.py
+++ b/tests/test_navigation.py
@@ -1,4 +1,4 @@
-from copy import deepcopy
+from copy import copy
 from pathlib import Path
 
 from mkdocs.commands.build import build
@@ -46,6 +46,8 @@ def test_plugin_static_nav(config_plugin_static_nav):
     files = i18n_plugin.on_files(files, config)
     nav = get_navigation(files, config)
     nav = i18n_plugin.on_nav(nav, config, files)
+    env = config.theme.get_env()
+    env = i18n_plugin.on_env(env, config, files)
     i18n_plugin.on_post_build(config)
     #
     assert i18n_plugin.i18n_configs["en"]["nav"] == EN_STATIC_NAV
@@ -61,14 +63,16 @@ def test_plugin_translated_nav(config_plugin_translated_nav):
     files = i18n_plugin.on_files(files, config)
     nav = get_navigation(files, config)
     nav = i18n_plugin.on_nav(nav, config, files)
+    env = config.theme.get_env()
+    env = i18n_plugin.on_env(env, config, files)
     i18n_plugin.on_post_build(config)
     #
-    fr_config = deepcopy(i18n_plugin.i18n_configs["fr"])
+    fr_config = copy(i18n_plugin.i18n_configs["fr"])
     fr_config["nav"] = FR_TRANSLATED_NAV
     fr_nav = get_navigation(i18n_plugin.i18n_files["fr"], fr_config)
     assert i18n_plugin.i18n_navs["fr"].__repr__() == fr_nav.__repr__()
     #
-    en_config = deepcopy(i18n_plugin.i18n_configs["en"])
+    en_config = copy(i18n_plugin.i18n_configs["en"])
     en_config["nav"] = EN_TRANSLATED_NAV
     en_nav = get_navigation(i18n_plugin.i18n_files["en"], en_config)
     assert i18n_plugin.i18n_navs["en"].__repr__() == en_nav.__repr__()

--- a/tests/theme_overrides/content.html
+++ b/tests/theme_overrides/content.html
@@ -1,0 +1,3 @@
+Testing on_env hooks
+
+{% set test = "comments" | value_in_frontmatter(page.meta) %}

